src/Entity/User.php

```diff
@@ -117,6 +117,23 @@ public function getAssignments(): Collection
         return $this->assignments;
     }
 
+    /**
+     * @return Collection|Assignment[]
+     */
+    public function getCancellableAssignments(): Collection
+    {
+        $assignments = $this->getAssignments();
+
+        $list = new ArrayCollection();
+        foreach ($assignments as $assignment) {
```

Comment:
```
$assignments->filter(function(Assignment $assignment): bool { return $assignment->isCancellable()})
```

Reply:
Another thing, why don't you bring only the assignments that you need instead of brings everything and filter after?

Reply:
Another thing, did you check if this is getting the data from the cache layer instead of database?

Reply:
First comment - agree, would be changed.
Second - to be discussed about better place. But the intention is to get the assignments of _current user_ which can be cancelled. In any case we already have a "link" to _all_ assignments of the user inside user object.
Third - yes, checked, for me no additional queries were produced.